tests/tasks.nix

```diff
@@ -0,0 +1,67 @@
+#
+# This derivation builds a Makefile, which is then passed to gmake
+# in order to perform the necessary Continuous Testing (CT) tasks,
+# with correct dependency ordering and as much parallelism as
+# possible.
+#
+# A Makefile is used solely because we need access to the Nix store
+# (which contains cached artifacts from previous CT runs) in order
+# to get reasonable performance.  Since recursive-nix is neither
+# stable nor ready for use, a Makefile is the simple interim solution.
+#
+
+{ pkgs-path ? ../.
+, shortRev ? builtins.substring 0 8 (lib.commitIdFromGitRepo ../.git)
+, pkgs ? import pkgs-path {}
+, lib ? pkgs.lib
+}:
+
+let
+
+  # Attrset where the attrname is the target name and the attrvalue
+  # becomes the lines to be used.  No tab characters needed.
+  #
+  # These commands are executed "outside of" nix-build, so they have
+  # access to the shared store which will retain cached artifacts
+  # across CT runs.
+  #
+  targets = {
+    pkgs-test-release =''
+      ${pkgs.nix}/bin/nix-build ./pkgs/test/release
+    '';
+
+    # uncomment after ofborg merges and deploys https://github.com/NixOS/ofborg/pull/660
+    /*
+    libtests = ''
+      ${pkgs.nix}/bin/nix-build ./lib/tests/release.nix
+    '';
+    */
+  };
+
+in
+pkgs.mkShell {
+  pname = "nixpkgs-ct";
+  version = shortRev;
+
+  passAsFile = [ "makefile" ];
+  makefile = ''
+    ${lib.concatStrings
+      (lib.mapAttrsToList
+        (name: val:
+          ''
+          .PHONY: ${name}
+          default: ${name}
+          ${name}:
+          ${lib.concatStringsSep "\n"
+            (map (line: if line=="" then "" else "\t${line}")
+              (lib.splitString "\n" val))}
+          '')
+        targets)}
```

Comment:
Why generate a makefile? Couldn't we just generate a shell script? Let's keep it simple where possible

Reply:
I agree with @infinisil, and logic like this, if we need it should go in its own function (`lib.generators`?) and have tests.
Non-trivial test support code needs tests, just like any other code.

Reply:
> Why generate a makefile?

Parallelism.

`make` is the simplest tool that knows how to run tasks with dependencies in parallel.

Our ofborg-driven CT is already parallel, so if the single-entry-point tests are run serially any attempt to migrate the existing tests to the new entry point will cause a major usability regression.

Reply:
For now we don't need this though, there's only a single task. And the one you commented out for the future doesn't depend on it either. In fact, both of them are just derivations, so we can just use Nix itself to run them in parallel! The comment about these tasks running "outside of nix-build" doesn't seem to apply.